prov/rxm/src/rxm_conn.c

```diff
@@ -391,12 +395,31 @@ static void rxm_conn_wake_up_wait_obj(struct rxm_ep *rxm_ep)
 
 static int
 rxm_conn_handle_event(struct rxm_ep *rxm_ep, ssize_t rd, uint32_t event,
-		      struct fi_eq_cm_entry *entry, void *context)
+		      struct fi_eq_cm_entry *entry, struct fi_eq_err_entry *err_entry)
 {
 	struct rxm_cm_data *cm_data;
 
 	if (rd == -FI_ECONNREFUSED) {
-		ofi_cmap_process_reject(rxm_ep->util_ep.cmap, context);
+		enum util_cmap_reject_flag cm_reject_flag;
+
+		if (OFI_LIKELY(err_entry->err_data_size >=
+				sizeof(enum util_cmap_reject_flag))) {
+			assert(err_entry->err_data);
+			cm_reject_flag = *((enum util_cmap_reject_flag *)
+							err_entry->err_data);
+		} else {
+			FI_WARN(&rxm_prov, FI_LOG_FABRIC,
+				"Error -FI_ECONNREFUSED was provided without error data\n");
+			cm_reject_flag = CMAP_REJECT_GENUINE;
```

Comment:
Got it but I think we may have to use a separate version in the header instead of depending on prov_version. prov_version may change but cm_data headers / how we convey reject may not change.